src/partest-extras/scala/tools/partest/IcodeComparison.scala

```diff
@@ -48,8 +48,13 @@ abstract class IcodeComparison extends DirectTest {
     compile("-d" :: testOutput.path :: arg0 :: args.toList : _*)
     val icodeFiles = testOutput.files.toList filter (_ hasExtension "icode")
 
-    try     icodeFiles sortBy (_.name) flatMap (f => f.lines.toList)
-    finally icodeFiles foreach (f => f.delete())
+    // Some methods in scala.reflect.io.File leak an InputStream, leaving the underlying file open.
+    // Windows won't delete an open file, but we must ensure the files get deleted, since the logic
+    // here depends on it (collectIcode will be called multiple times, and we can't allow crosstalk
+    // between calls).  So we are careful to use `slurp` which does call `close`, and careful to
+    // check that `delete` returns true indicating successful deletion.
+    try     icodeFiles sortBy (_.name) flatMap (f => f.slurp().lines.toList)
+    finally icodeFiles foreach (f => require(f.delete()))
```

Comment:
As a matter of fact, I recall that the fix to slurp was because I ran tests on windows and hit the same issue.

https://github.com/scala/scala/blob/v2.11.7/src/reflect/scala/reflect/io/Streamable.scala#L112

Li Haoyi just complained about the behavior of the other methods (he was using Source I think) and I checked that indeed there are leakages, which is too bad.

OTOH, at the Java office, I just had to verify that when using try-with-res, you have to close wrapped resources or be very sure about the behavior. End of this article points out that not all close methods guarantee to close decorated resources.

http://www.oracle.com/technetwork/articles/java/trywithresources-401775.html

So basically, Damn you all to hell! [PAN to reveal STATUE OF LIBERTY half-buried in the sand.]